macros: ungate `routes`/`url_patterns` re-exports so #4132's cross-target emission is reachable from wasm consumers

## Summary

PR #4132 made `#[routes]`-generated per-app client URL accessors emit on `wasm32-unknown-unknown`, but the macro re-exports themselves are still gated to native:

`src/lib.rs:307-309`

```rust
#[cfg(native)]
pub use reinhardt_macros::routes;
#[cfg(native)]
pub use reinhardt_macros::url_patterns;
```

Because the re-export is gated, any downstream module that contains `#[reinhardt::routes]` or `#[reinhardt::url_patterns]` cannot be compiled for wasm32 — which means the file containing the macro invocation must be marked `#[cfg(native)]`, which means the generated `ResolvedUrls` struct (intentionally cross-target after #4132) is unreachable on wasm anyway.

This is the missing half of #4119: #4132 fixed the macro **emission** but the macro **invocation gate** at the framework boundary defeats it.

## Reproduction

In a downstream consumer (e.g. `reinhardt-cloud` PR #528 / branch `refactor/issue-519-typed-urls-client-accessor`):

```rust
// dashboard/src/config/urls.rs (cross-target attempt)
use reinhardt::routes;
use reinhardt::urls::prelude::UnifiedRouter;

#[routes]
pub fn routes() -> UnifiedRouter {
	UnifiedRouter::new()
		.mount_unified("/", crate::apps::dashboard::urls::url_patterns())
}
```

Build:
```sh
cargo check -p reinhardt-cloud-dashboard --target wasm32-unknown-unknown --lib
```

Result:
```
error[E0432]: unresolved import `reinhardt::url_patterns`
note: found an item that was configured out
   --> reinhardt-web/src/lib.rs:309:27
    |
308 | #[cfg(native)]
309 | pub use reinhardt_macros::url_patterns;
```

The same error occurs for `reinhardt::routes` if the consumer tries to mark its `#[routes]` site cross-target.

## Impact

Without ungating the re-exports, `ResolvedUrls::from_global().client().<app>().<route>()` is only reachable from native call sites, so cross-target SPA call sites (e.g., `client/layout.rs`, `apps/<app>/client/pages/*.rs`) still need a runtime untyped resolver shim. This forces every reinhardt-web SPA consumer to maintain a `url_for_spa()`-style workaround module — exactly the situation #4119 / #4132 set out to eliminate.

## Suggested fix

Remove the `#[cfg(native)]` guards on the macro re-exports, since the macros are proc-macros that run host-side at compile time and #4132 already made their emission cross-target:

```diff
-#[cfg(native)]
 pub use reinhardt_macros::routes;
-#[cfg(native)]
 pub use reinhardt_macros::url_patterns;
```

If individual macros still emit native-only artifacts in some configurations, those gates belong inside the macro, not at the re-export site.

## Related

- Closes blocker for: kent8192/reinhardt-cloud#519, kent8192/reinhardt-cloud#528, kent8192/reinhardt-cloud#534
- Refs #4119 (per-app accessors not generated on wasm — fixed by #4132)
- Refs #4132 (the cross-target emission this issue's fix would unlock)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -272,13 +272,16 @@ reinhardt-urls = { workspace = true }
 
 reinhardt-core = {workspace = true, features = ["types"]}
 
+# Proc-macro crate: runs host-side at compile time, safe on every target
+# (re-exported as `reinhardt::routes` / `reinhardt::url_patterns` for cross-target consumers)
+reinhardt-macros = { workspace = true }
+
 # Server-side dependencies (NOT for WASM)
 [target.'cfg(not(all(target_family = "wasm", target_os = "unknown")))'.dependencies]
 # Core dependencies
 reinhardt-utils = { workspace = true }
 reinhardt-views = { workspace = true }
 reinhardt-http = { workspace = true }
-reinhardt-macros = { workspace = true }
 async-trait = { workspace = true }
 bytes = { workspace = true }
 hyper = { workspace = true }

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -303,9 +303,7 @@ macro_rules! define_views {
         $crate::flatten_imports!($($tt)*)
     };
 }
-#[cfg(native)]
 pub use reinhardt_macros::routes;
-#[cfg(native)]
 pub use reinhardt_macros::url_patterns;
 #[cfg(native)]
 pub use reinhardt_macros::viewset;

